File: src/org/erusu/leetcode/problemList/_001_TwoSum/TwoSum.cpp
#include <map>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::map<int, int> * numLocations = new std::map<int, int>();
        for(int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if(numLocations->find(complement) != numLocations->end())
                return {numLocations->at(complement), i};
            numLocations->insert(std::pair<int, int>(nums[i], i));
        }
        return {};
    }
};

File: src/org/erusu/leetcode/problemList/_001_TwoSum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        numLocations = {};
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in numLocations:
                return [numLocations[complement], i]
            else:
                numLocations[nums[i]] = i;
        return []